environments/wac-syntax.md

```diff
@@ -0,0 +1,204 @@
+---
+title: "Workspaces As Code Syntax"
+description: "Details on Workspaces As Code fields"
+state: beta
+---
+
+## About YAML syntax
+
+Workspaces As Code config files use YAML syntax, typically ending with the
+`.yml` or `.yaml` file extension. In order to create an environment from a
+config file you must save the file to `.coder/coder.yaml` relative to the root
+of your repository.
+
+## Config
+
+The following demonstrates a fully populated WAC config file. For more detailed
+information on select fields [see below](#config-fields).
+
+```yaml
+version: 0.1
+workspace:
+  type: kubernetes
+  spec:
+    image: index.docker.io/ubuntu:18.04
+    container-based-vm: true
+    cpu: 4
+    memory: 16
+    disk: 128
+    gpuCount: 1
+    labels:
+      com.coder.custom.hello: "hello"
+      com.coder.custom.world: "world"
+    tolerations:
+      - key: my-key
+        operator: Equal
+        value: my-value
+        effect: NoExecute
+        tolerationSeconds: 3600
+  configure:
+    - name: "install curl"
+      run: |
+        apt update
+        apt install -y curl
+    - name: "install Go binary"
+      run: "go install"
+      directory: /home/coder/go/src/github.com/my-project
+      shell: "bash"
+      env:
+        GOPATH: /home/coder/go
+```
+
+## Config Fields
+
+### version
+
+The version of the config file being used. The current version is `0.1`.
```

Comment:
Is the version number something the user uses to distinguish between config file variants, or does it correspond to something in Coder?

Reply:
config file variants.

If we have to break the yaml structure, we will bump the version number. 
Eg: Renaming fields

Reply:
It's to distinguish between config file variants.